manifests/auto_cloud_swarm.pp

```diff
@@ -0,0 +1,68 @@
+# == Class: robottelo_slave::auto_cloud_swarm
+#   Automatically setup everything needed to get a host from blank OS install up
+#   to a fully functioning slave registered to Jenkins with the swarm plugin
+#
+# === Parameters
+#
+# [*jenkins_master*]
+#   The jenkins master server to attach to
+#
+class robottelo_slave::auto_cloud_swarm(
+  $jenkins_master = undef,
+  $jenkins_user   = undef,
+  $jenkins_pwd    = undef,
+  $jenkins_labels = 'builders',
+  $slave_user     = 'jenkins',
+) {
+  yumrepo { 'epel-temp':
+    mirrorlist => 'https://mirrors.fedoraproject.org/metalink?repo=epel-7&arch=$basearch',
+    gpgcheck   => 0,
+    enabled    => 1,
+  } ->
+  package {'epel-release':
+    ensure => 'latest',
+    before => [
+      Package['mock', 'tito', 'git-annex', 'koji', 'npm'],
+      Class['robottelo_slave'],
+    ],
+  }
+
+  ensure_resource('group', $slave_user, { 'ensure' => 'present' })
+  ensure_resource('user', $slave_user, { 'ensure' => 'present' })
+  User<| title == $slave_user |> {
+    managehome => true,
+    gid        => $slave_user,
+    groups     => [$slave_user, 'mock'],
+    require    => Package['mock'],
+  }
+  User[$slave_user] -> Class['robottelo_slave']
+
+  include ::robottelo_slave::brew
+  ensure_packages(['mock'])
```

Comment:
OK - I wasn't sure if this class should strictly handle configuring the jenkins slave aspect and the "builder" related packages etc. should be covered separately.